tests/RetryAnnotationTraitTest.php

```diff
@@ -17,145 +17,139 @@ class RetryAnnotationTraitTest extends TestCase
+     * @expectedException \InvalidArgumentException
      * @expectedExceptionMessage The @retryAttempts annotation requires an integer as an argument
      */
-    public function testNoArgumentToRetryAttemptsAnnotation()
+    public function testNoArgumentToRetryAttemptsAnnotation(): void
     {
         $this->getRetryAttemptsAnnotation();
     }
 
     /**
-     * @expectedException LogicException
+     * @expectedException \InvalidArgumentException
      * @expectedExceptionMessage The @retryAttempts annotation requires an integer as an argument
      */
-    public function testEmptyStringToRetryAttemptsAnnotation()
+    public function testEmptyStringToRetryAttemptsAnnotation(): void
     {
-        $this->validateRetryAttemptsAnnotation('');
+        $this->parseRetryAttemptsAnnotation('');
     }
 
     /**
-     * @expectedException LogicException
+     * @expectedException \InvalidArgumentException
      * @expectedExceptionMessage The @retryAttempts annotation must be an integer but got "'foo'"
      */
-    public function testInvalidStringArgumentTypeToRetryAttemptsAnnotation()
+    public function testInvalidStringArgumentTypeToRetryAttemptsAnnotation(): void
     {
-        $this->validateRetryAttemptsAnnotation('foo');
+        $this->parseRetryAttemptsAnnotation('foo');
     }
 
     /**
-     * @expectedException LogicException
+     * @expectedException \InvalidArgumentException
      * @expectedExceptionMessage The @retryAttempts annotation must be an integer but got "1.2"
      */
-    public function testInvalidFloatArgumentTypeToRetryAttemptsAnnotation()
+    public function testInvalidFloatArgumentTypeToRetryAttemptsAnnotation(): void
     {
-        $this->validateRetryAttemptsAnnotation('1.2');
+        $this->parseRetryAttemptsAnnotation('1.2');
     }
 
     /**
-     * @expectedException LogicException
+     * @expectedException \InvalidArgumentException
      * @expectedExceptionMessage The @retryAttempts annotation must be 0 or greater but got "-1"
      */
-    public function testNonPositiveIntegerToRetryAttemptsAnnotation()
+    public function testNonPositiveIntegerToRetryAttemptsAnnotation(): void
     {
-        $this->validateRetryAttemptsAnnotation(-1);
+        $this->parseRetryAttemptsAnnotation(-1);
     }
 
-    public function testValidRetryAttemptsAnnotations()
+    public function testValidRetryAttemptsAnnotations(): void
     {
-        $this->assertEquals(0, $this->validateRetryAttemptsAnnotation(0));
-        $this->assertEquals(0, $this->validateRetryAttemptsAnnotation('0'));
-        $this->assertEquals(1, $this->validateRetryAttemptsAnnotation(1));
-        $this->assertEquals(1, $this->validateRetryAttemptsAnnotation('1'));
-        $this->assertEquals(1, $this->validateRetryAttemptsAnnotation(1.0));
-        $this->assertEquals(1, $this->validateRetryAttemptsAnnotation('1.0'));
+        $this->assertEquals(0, $this->parseRetryAttemptsAnnotation('0'));
```

Comment:
Since I added the strict types and in practice the values passed are only strings, I removed the tests with integer/float values